crates/core/src/credits.rs

```diff
@@ -0,0 +1,113 @@
+use std::{collections::HashMap, io::prelude::*, path::PathBuf};
+
+use hub_core_schemas::{CreditsEvent, CreditsEventKey};
+
+use crate::{prelude::*, producer, util::DebugShim};
+
+impl producer::Message for CreditsEvent {
+    type Key = CreditsEventKey;
+}
+
+/// Service startup configuration for charging credits and reading the credit
+/// sheet
+#[derive(Debug)]
```

Comment:
```suggestion
#[derive(Debug, Clone)]
```
need the credits client to implement Clone